### ISSUE TITLE
Add option to do a secrets apps reset for the nk3 similar to: nitropy nk3 secret reset 

### ISSUE THREAD

**Issue by nestire** [Nitrokey/nitrokey-hotp-verification#42]
See #41 for more  [detail](https://github.com/Nitrokey/nitrokey-hotp-verification/issues/41#issuecomment-2481898790
)

this should work with user presence (as the nitropy version) so no nk3 firmware change is need. 

bigger discussion ticket regarding this is #36

**Comment by daringer** [comment]
This leads to a additional commandline argument:
```
hotp_verification reset 
```
This will:
* factory-reset the secrets app
* only work for nitrokey 3
* silently fail (to be consistent with `change-pin` and #40) for non nitrokey 3s

**Comment by tlaurion** [comment]
Ideally, `hotp_verification reset` would accept a parameter here, which would be the secret app PIN.

Otherwise as can be seen under https://github.com/linuxboot/heads/pull/1850/commits/07f3710f7847c07b29da249cc36c810c41ea7124, it requires an additional step which would be PIN change, but we don't have a PIN here, since we just reset :)


I still think, as said under https://github.com/Nitrokey/nitrokey-hotp-verification/issues/36#issuecomment-2479284125 that changing PIN is not really important to Heads use case since we reset. What is important is setting a secret app PIN at moment of oem factory reset/re-ownership.


Need:

- [ ] `hotp_verification reset DESIRED_SECRET_APP_PIN`

**Comment by daringer** [comment]
ok, so now that things are implemented you bring up the idea to change the requirements ? 
awesome, now the plan change after it is already implemented is:

there should be a new command:

```
hotp_verification reset <SECRET_PIN>
```

This will:

*  factory-reset the secrets app
* only work for nitrokey 3
* silently fail 
* the `SECRET_PIN` is required


Further we will drop the implementation and requirement described in [this issue](https://github.com/Nitrokey/nitrokey-hotp-verification/issues/40) ??

**Comment by tlaurion** [comment]
>> For feature freeze, we need re-ownership to work as intended so that OEM can seal HOTP and end user can define new secrets per re-ownership.
 

>> TLDR: #41>#38>#39, nice to have #40

>#41 was deemed impossible because physical presence removal required nk3 firmware change, and led to #42

> So priorities are:

> #42>#41>#38>#39, nice to have #40

>_Originally posted by @tlaurion in https://github.com/Nitrokey/nitrokey-hotp-verification/issues/36#issuecomment-2506996805_


---

>@nestire then another issue needs to be opened seperating need for factory reset of secure element of physical presence. 

>nitropy nk3 secret reset equivalent should be implemented into hotp-verification, which would not be bound to physical presence and not bound to new firmware version.

> Heads needs to be able to do oem factory reset of htop (seal secret prior of shipping), and re-ownership needs to be able to reset that, just like gpg pins.

> For feature freeze: oem will not have a completely unattended experience provisioning randomized secrets because of lack of firmware update, but at least they will be able to seal that secret, and end users be able to reown that part as expected without nk3 being a regression as compared to nk2 for transfer of ownership.

> _Originally posted by @tlaurion in https://github.com/Nitrokey/nitrokey-hotp-verification/issues/41#issuecomment-2481898790_


----

> ok, so now that things are implemented you bring up the idea to change the requirements ?
> awesome, now the plan change after it is already implemented is:
> `hotp_verification reset <SECRET_PIN>`


@nestire : nk3 didn't respect heads use case since its inception: nk3 never respected requirements. I won't participate in cheap politics nor will feel bad to point wrong doing. Not your fault, dunno which fault and it doesn't matter. What matters is reducing friction, making security accessible and not changing security promises, which were all broken with nk3 and i'm still helping you guys fix your mess. Should I stop?


>> Further we will drop the implementation and requirement described in [this issue](https://github.com/Nitrokey/nitrokey-hotp-verification/issues/40) ? Please close this issue, if it is not needed anymore it was created due to your request(s).

> #42>#41>#38>#39, nice to have #40

I always said #40 was nice to have, #41 was my need, #38 and #39 are simply misleading users, but I let Nitrokey deal with that with unneeded support requests, problem is that everyone depends on nk3, and everyone answers to repeating support requests, PINs that don't exist, set PIN silently and even replaced remote attestation sealing by physical presence alone before nk3 firmware 1.7.1.  Nitrokey changed the rules of the game, I'm just trying to bring back things as they should be.

TLDR:  
gpg when doing factory reset does for nk2/nk2 pro/lb:
- set GPG User PIN to 123456
- set GPG User PIN to 12345678
- prior of nk3, GPG User PIN was setting nk3 equivalent of SECRET_APP_PIN
- nk3 designers decided to 
  - fake existence of a GPG User PIN instead of marking it a single SECRET_APP_PIN
  - fake existence of a GPG Admin PIN instead of marking it a single SECRET_APP_PIN
- nk3 decided to replace HOTP sealing with PIN with physical presence prior of nk3 firmware 1.7.1
- nk3 decided that it was not important to respect prior gpg provisioned Admin PIN, same used for remote attestation with HOTP (secret app PIN) leaving it unset
- nk3 decided that sealing and setting prior GPG Admin PIN used on prior version was going to be silently set on first use, leaving to user error and incomprehensible support requests that I tackled on first iteration, pinging with links when it reoccured and recently tagged nitrokey directly because nitrokey won't care/fix.
- I worked more then 20 hours with the help of QubesSO team to what lead to lat hotp_verification temporary fix and nk3 firmware 1.7.1 wich was reeleased without waiting for partners to also do the same with their clients
- I'm again iterating and explaining where Nitrokey doesn;t listen and still want double useless ADMIN PIN/User PIN for secure elemenet, wants to report firmware versions that don't exist and need to be reminded of what exissted and weorked prior of nk3 which nitrokey should have made sure to not break compatibility.

@jan23: that's enough. Maake things right, stop asking me for words you don't read.It's going to be journalists next, seriously.
Reminder: using a nk3 is not a novelty product. Its is used for remote attestation. Users need to understand what PIN is set used when, and we need to move together as an ecosystem. It's not my fault if you work in silo and I discover bugs in your implementation, nor if Nitrokey decides to not follow prior working code and not do regression tests, and if  Nitrokey never communicated that PIN was not needed anymore to seal HORTP (basically permitting any PIN to be entered prior of nk3 1.7.1!!!)

Make things right. Ask questiosn when unsure. Communicate with your users. Make hotp_verification output useful and actinnable info. MAke hotp_verification secret set similar to gpg. Don't ask me how to do your work: just don't break what worked previously, in other words: do regression testing, otherwise this hurts the whole ecosystem aand noi, I do not have unlimited unpaid time, again, to fix mess that was already discussed many times and at many places. This is to fix properly what wasn't fixed under nk3 1.7.1 and hotp_verification 1.6. 

Nothing new, was discussed before in simple terms "don't create regression as compared to nk2" from a Heads perspective.

**Comment by daringer** [comment]
is my previous description of the feature you requested as you would like to have it?

**Comment by tlaurion** [comment]
> is my previous description of the feature you requested as you would like to have it?

@sosthene-nitrokey already acknowledged at https://github.com/Nitrokey/nitrokey-hotp-verification/pull/46#issuecomment-2508143760

PR are for implementation details. Requirements are: don't break nk2 compatibility otherwise open or under heads for things nk3 changed. Thank you.

**Comment by tlaurion** [comment]
> there should be a new command:
> 
> ```
> hotp_verification reset <SECRET_PIN>
> ```
> 
> This will:
> 
> * factory-reset the secrets app
> * only work for nitrokey 3
> * silently fail
> * the `SECRET_PIN` is required

Why silently fail? why not only work for nk3? Anyway, PoC under https://github.com/linuxboot/heads/pull/1850/files#diff-bec315d56e315e18809e58210db23903cfa07559d297a1910f2c3151f6e6e97dR141-R149 only applying to nk3.

But yes, @daringer. This would make it compliant to nk3 with GPG Admin PIN being set prior of use under Heads use case.

**Comment by daringer** [comment]
I would love to read a clear "yes" or "no", if I am asking a closed (yes/no) question. 
Interpreting your comment, I infer that you say "yes"

**Comment by tlaurion** [comment]
@daringer : reset should put a default pin, as gpg Admin PIN did per gpg. That would be 12345678. As per prior of nk3.

This pin should be Secure App PIN, not have a non existing admin/user PIN to not mislead users, UX, and flood support requests because non-sensical and.. not existing.

The clear yes that you are asking is:
If reset doesn't set a default secure element pin and nitrokey obstine into keeping things as they are, then yes, I will deal with `hotp_verification reset ADMIN_PIN`, but warning of this being a patch upon bad design and reinventing the wheel and complexifying things.

We code but don't think here. I can only repeat myself. This is tiring. 

- `hotp_verification reset` should set a default pin as for nk2 through gpg toolstack, if you want to implement a pin change. Otherwise I don't understand the game were playing here. 

- `hotp_verification reset SECRET_APP_PIN` I can work with, but don't understand your design decisions and thought process and design decisions. 

TLDR : `hotp_verification reset SECRET_APP_PIN` is bare minimal with physical presence constraints of today. Heads won't change the PIN and have a different workflow of gpg PINs. Hope the secure app PIN will be the wording used, otherwise I'll redirect all support requests to nitrokey for that matter in the future.
